packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -1164,4 +1169,52 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         return itemA.localeCompare(itemB);
     }
 
+    /**
+     * @param recursive if true, all child nodes will be included in the stringified result.
+     */
+    nodeToString(node: TreeNode, recursive: boolean): string {
+        if (SearchInWorkspaceFileNode.is(node) || SearchInWorkspaceRootFolderNode.is(node)) {
+            if (recursive) {
+                return this.nodeIteratorToString(new TopDownTreeIterator(node, { pruneSiblings: true }));
+            }
+            return this.labelProvider.getLongName(node.uri);
+        }
+        if (SearchInWorkspaceResultLineNode.is(node)) {
+            return `  ${node.line}:${node.character}: ${node.lineText}`;
+        }
+        return '';
+    }
+
+    treeToString(): string {
+        return this.nodeIteratorToString(this.getVisibleNodes());
+    }
+
+    protected *getVisibleNodes(): IterableIterator<TreeNode> {
+        for (const { node } of this.rows.values()) {
+            yield node;
+        }
+    }
+
+    protected nodeIteratorToString(nodes: Iterable<TreeNode>): string {
+        const strings = [];
+        for (const node of nodes) {
+            const string = this.nodeToString(node, false);
+            if (string.length !== 0) {
+                strings.push(string);
+            }
+        }
+        return strings.join(isWindows ? '\r\n' : '\n');
+    }
+}
+
+export namespace SearchInWorkspaceResultTreeWidget {
+    export namespace Menus {
+        export const base = ['siw-tree-context-menu'];
+        /** Dismiss command, or others that only affect the widget itself */
+        export const internal = [...base, '1_internal'];
+        /** Copy a stringified representation of content */
+        export const copy = [...base, '2_copy'];
+        /** Commands that lead out of the widget, like revealing a file in the navigator */
+        export const external = [...base, '3_external'];
```

Comment:
minor - should we make the const uppercase:

```suggestion
        export const BASE = ['siw-tree-context-menu'];
        /** Dismiss command, or others that only affect the widget itself */
        export const INTERNAL = [...base, '1_internal'];
        /** Copy a stringified representation of content */
        export const COPY = [...base, '2_copy'];
        /** Commands that lead out of the widget, like revealing a file in the navigator */
        export const EXTERNAL = [...base, '3_external'];
```